### ISSUE TITLE
Add default constructors for CUDA/HIP/omptarget reducers.

### ISSUE THREAD

**Issue by rchen20** [LLNL/RAJA#820]
The current CUDA/HIP/omptarget reducers omit default constructors because these objects need to be constructed on the device. The seq and openmp reducers can be default constructed (on the host). Should we attempt to support default construction of the CUDA/HIP/omptarget reducers to make the API consistent?

**Comment by rhornung67** [comment]
Interesting that you added this issue. I ran into this while working through the reduction tests.  This shouldn't be too hard, right? We would define default values in the same way we do for the CPU backends and initialize the device objects the same way we do now when calling a non-default ctor.

**Comment by rchen20** [comment]
Right, it shouldn't be a big change. In fairness, @MrBurmark brought this up when he noticed that my reducer unit tests did not call the default constructors of CUDA/HIP/omptarget. The only questionable default reducer value would be for min reductions. We could default to max values for primitive types, but maybe the user should be forced to initialize to their preferred value?

**Comment by rhornung67** [comment]
I agree that users should have to define initalization values for their own types.

**Comment by rhornung67** [comment]
https://github.com/llnl/RAJA/pull/2061 made reducer construction consistent with respect to initial reduction values. The v2026.07.0 release notes state that default initialization for RAJA reducers is deprecated. In the next major release, users will be required to explicitly provide initial values.
